archunit/src/test/java/com/tngtech/archunit/core/domain/JavaTypeVariableTest.java

```diff
@@ -0,0 +1,50 @@
+package com.tngtech.archunit.core.domain;
+
+import java.io.Serializable;
+
+import com.tngtech.archunit.core.importer.ClassFileImporter;
+import org.junit.Test;
+
+import static com.tngtech.archunit.testutil.Assertions.assertThat;
+
+public class JavaTypeVariableTest {
```

Comment:
Hmm, originally I thought I would only cover the parts here that are not covered yet. I.e. there is no test gap for the other methods in `ClassFileImporterGenericClassesTest`, you cannot destroy these methods and yet have `CFIGCT` pass. So I'm really not a big fan of testing the same thing in multiple tests. It would be okay for me if

a) the other test would be really slow so it makes sense to gather a quicker feedback even though the other test would test it, too
b) I would use a more "skinny" approach to set up the test fixture, but I don't think this makes sense here, because creating the `JavaTypeVariable` through some stub import provides little value and makes the test more brittle and less realistic

I do see your point that it is somewhat confusing to have only these tests here and I think this will also confuse people in the future about where to add tests :thinking: 

I see two ways to go forward:

a) Add some comment like "besides toString() the behavior of JavaTypeVariable is already covered in ClassFileImporterGenericClassesTest"
b) add the `toString` assertion to `JavaTypeVariableAssertion`

I think the reason this ended up in a separate test is that I consider `toString` of a different quality from the other API. It is pretty much only a debug output, no parts of ArchUnit rely on it and the output is also nothing I would really consider as a format to be used by an end user. On the other hand it's still public API and not trivial in this case so I felt better to add a test. That's the history why it ended up there.

Given all that, what do you think? :wink:

Reply:
From a documentation point of view, a short reference from `JavaTypeVariableTest` to `ClassFileImporterGenericClassesTest` – i.e. your suggested way a) to go forward – would certainly be helpful. 👍 (In order to understand the semantics of some class, I sometimes look at its tests to learn how it can be used and how it works. In this case, one could only find the `.toString()` tests immediately.)

Due to the power (and thus complexity) of `JavaTypeAssertion`/`JavaTypeVariableAssertion`, `ClassFileImporterGenericClassesTest` is not the most simple code to learn what a `JavaTypeVariable` specifically is.

So I personally wouldn't mind to test (different aspects of) the same feature in multiple tests (especially if it's a public API and not some implementation detail that might change): 
* In `JavaTypeVariableTest` more from the unit test perspective _"Given a single `JavaTypeVariableTest`, these are the properties", only using simple assertions.
* In `ClassFileImporterGenericClassesTest` more from the integration test perspective _"Given a generic class, these are its type variables"_.

But these are just some thoughts. A comment linking to `ClassFileImporterGenericClassesTest` is certainly good enough.

Reply:
Okay, I see your point :+1: I also saw that `JavaTypeTest` kind of mixes tests for different subclasses, so I split it between the tests. And added some simple tests for the public API.